### PR TITLE
Remove old-style classes code, remove check for new-style class

### PR DESCRIPTION
**Pull request by akamat10**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description
* Given the removal of old-style class checks in astroid and `newstyle` attribute for `ClassDef` instances committed in pylint-dev/astroid#2561, cleaning up pylint code related to that change. 
* Also, change exceptions code/doc wording to remove `new style` wording since there is only one type of class in Python 3 and it has been around for a long time.

Refs pylint-dev/astroid#2561

### CODE REVIEW

File: tests/functional/i/invalid/invalid_exceptions/invalid_exceptions_raised.py [hunk 1]
```diff
@@ -1,17 +1,13 @@
 # pylint:disable=too-few-public-methods,import-error,missing-docstring, not-callable, import-outside-toplevel
-"""test pb with exceptions and old/new style classes"""
```
Comment (Pierre-Sassoulas) [inline tests/functional/i/invalid/invalid_exceptions/invalid_exceptions_raised.py:2]: Let's not remove the test cases.

**Review by Pierre-Sassoulas — DISMISSED** [review]

**Review by DanielNoord — DISMISSED** [review]
Let's remove them Pierre. `OldStyleClass` and `NewStyleClass` are no different

File: tests/functional/i/invalid/invalid_exceptions/invalid_exceptions_raised.py (hunk 1 above)
Comment (akamat10) [reply tests/functional/i/invalid/invalid_exceptions/invalid_exceptions_raised.py:2]: The tests are duplicates and so I removed them.

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for this. Mind resolving the lints?

**Comment by akamat10** [comment]
Thanks for the reviews! Do I need to fix the code coverage failure, too? This PR is removing code.

**Comment by jacobtylerwalls** [comment]
No worries on the coverage check, but we do want the docs check to pass: see the git diff it outputs.

**Comment by akamat10** [comment]
Thanks! Hopefully, that fixes it. Should `Checks / documentation (pull_request)` be made required to give better feedback to the person submitting PR?

**Comment by akamat10** [comment]
Do I need to worry about the messages on https://github.com/pylint-dev/pylint/pull/9925#issuecomment-2354319946 ? Is it saying they are caused by my changes?

**Comment by Pierre-Sassoulas** [comment]
I don't think the primer are due to this, they run on the latest commit so when a commit is introduced in the repository we prime this can happen. (#8998)

**Review by jacobtylerwalls — APPROVED** [review]

**Comment by jacobtylerwalls** [comment]
> when a commit is introduced in the repository we prime this can happen

I think it's more likely there are unknown bugs in pylint that cause the primer noise; it happens far too frequently for it to be attributed to a project commit changing its packaging to pyproject.toml or whatever happened in that issue.

**Comment by akamat10** [comment]
> > when a commit is introduced in the repository we prime this can happen
> 
> I think it's more likely there are unknown bugs in pylint that cause the primer noise; it happens far too frequently for it to be attributed to a project commit changing its packaging to pyproject.toml or whatever happened in that issue.

May be yeah. The Django errors don't seem to be bugs in Django code.

**Comment by Pierre-Sassoulas** [comment]
Yeah, the no-member for django are expected if we don't use pylint-django. We might want to to our internal own conf for primed project not using pylint that really need a plugin.

**Comment by jacobtylerwalls** [comment]
(It's the indeterminacy that's more worrying to me than that.)

**Comment by akamat10** [comment]
I don't fully understand how the github actions are setup but I think it might be something to do with setting the `cancel-in-progress` for some of these github actions. Is it possible if commits are sent in quick succession, the partially completed workflow is terminated while it is installing in the virtual environment and when we check the cache to see if the virtual environment already exists, we are picking up a partially created virtual environment. It seems quite a few of the errors are tied to what pylint dependencies (direct or transient) have been installed even if they don't appear to be at first glance. The `redefined-variable-type` in https://github.com/pylint-dev/pylint/pull/9925#issuecomment-2354194223 is tied to whether `colorama` has been installed in the virtual environment.

**Comment by jacobtylerwalls** [comment]
I don't think so, see #9309 for a reproduction.

**Comment by akamat10** [comment]
May be you found a different issue. I was able to reproduce my issue with and without colorama which made me think we have some bug tied to caching in the workflow. Will try to understand it more.

**Comment by jacobtylerwalls** [comment]
Got it, thanks for the investigation. As a debug tool to confirm this I can open a PR later today to fail primer jobs if the full dependencies are not installed 👍🏻

**Pull request merged**